src/doctokenizer.h

```diff
@@ -194,6 +194,9 @@ class DocTokenizer
     void pushState();
     void popState();
 
+    int  getState();
```

Comment:
I don't think these are needed, one can use `pushState()` and `popState()` instead, which also avoids "leaking" the state information to the caller.